File: scripts/verify-image-asset-index.py
import io, os, sys, tempfile, json
from pathlib import Path

root = Path(__file__).resolve().parents[1]
os.environ["FENGVOICE_DB_PATH"] = str(Path(tempfile.gettempdir()) / "fengvoice-phase2b-test.db")
sys.path.insert(0, str(root / "services" / "api"))

from fastapi.testclient import TestClient
from main import app

client = TestClient(app)

resp = client.post("/api/uploads/images", files={"file": ("paste.png", io.BytesIO(b"\x89PNG\r\n\x1a\n"), "image/png")})
assert resp.status_code == 200, resp.text
payload = resp.json()
assert "image_id" in payload, f"Response must include image_id: {payload}"
assert payload["image_id"].startswith("img_"), f"image_id prefix wrong: {payload}"
assert payload["url"].startswith("http://testserver/uploads/notes/"), payload

idx = root / "runtime" / "asset-index" / "note-images.jsonl"
assert idx.exists(), f"Index file missing: {idx}"
lines = idx.read_text(encoding="utf-8").strip().split("\n")
assert len(lines) > 0, "Index file empty"
last = json.loads(lines[-1])
assert last["image_id"] == payload["image_id"], f"image_id mismatch: {last['image_id']} != {payload['image_id']}"
assert last["source"] == "note_paste", f"source mismatch: {last['source']}"
assert len(last["sha256"]) == 64, f"sha256 length wrong: {len(last['sha256'])}"
assert last["mime_type"] == "image/png", f"mime_type mismatch: {last['mime_type']}"
assert last["size_bytes"] > 0, f"size_bytes not positive: {last['size_bytes']}"
assert last["image_id"].startswith("img_"), f"stored image_id prefix wrong: {last['image_id']}"

print("image asset index verification passed")

File: scripts/verify-image-asset-index.js
const assert = require("node:assert/strict");
const { spawnSync } = require("node:child_process");
const { existsSync, readFileSync } = require("node:fs");
const { resolve } = require("node:path");

const root = resolve(__dirname, "..");

// 1. Check asset_index.py exists
const assetIndexPath = resolve(root, "services/api/asset_index.py");
assert.ok(existsSync(assetIndexPath), "asset_index.py must exist");
const source = readFileSync(assetIndexPath, "utf8");
assert.match(source, /create_note_image_record/, "asset_index.py must export create_note_image_record");
assert.match(source, /sha256/, "asset_index.py must compute sha256");
assert.match(source, /image_id/, "asset_index.py must generate image_id");
assert.match(source, /jsonl/, "asset_index.py must write JSONL");

// 2. Check main.py returns image_id
const mainSource = readFileSync(resolve(root, "services/api/main.py"), "utf8");
assert.match(mainSource, /image_id/, "main.py must return image_id in upload response");

// 3. Check UploadedImage type
const apiSource = readFileSync(resolve(root, "apps/web/src/api.ts"), "utf8");
assert.match(apiSource, /image_id:\s*string/, "UploadedImage type must include image_id");

// 4. Check .gitignore
const gitignore = readFileSync(resolve(root, ".gitignore"), "utf8");
assert.match(gitignore, /runtime\//, ".gitignore must ignore runtime/");

// 5. Run Python backend test
const pyResult = spawnSync("python", [resolve(root, "scripts/verify-image-asset-index.py")], {
  cwd: root,
  encoding: "utf8",
  timeout: 30000,
});
if (pyResult.status !== 0) {
  process.stdout.write(pyResult.stdout);
  process.stderr.write(pyResult.stderr);
  process.exit(pyResult.status ?? 1);
}

console.log("image asset index verification passed");
